Sources/PortalSwift/Core/Mpc/PortalMpc.swift

```diff
@@ -268,9 +325,123 @@ public class PortalMpc {
       throw MpcError.unexpectedErrorOnEject("Unable to find address in addresses map.")
     }
 
+    print("Private Key: \(eip155PrivateKey)")
+
     return eip155PrivateKey
   }
 
+  public func ejectSolanaWallet(
+    _ method: BackupMethods,
+    withCipherText: String? = nil,
+    andOrganizationBackupShare: String? = nil,
+    usingProgressCallback _: ((MpcStatus) -> Void)? = nil
+  ) async throws -> String {
+    if self.version != "v6" {
+      throw MpcError.backupNoLongerSupported("[PortalMpc] Eject is no longer supported for this version of MPC. Please use `version = \"v6\"`.")
+    }
+
+    var cipherText = withCipherText
+    var organizationShare = andOrganizationBackupShare
+
+    guard let storage = self.backupOptions[method] else {
+      throw MpcError.unexpectedErrorOnEject("Backup method \(method.rawValue) not registered.")
+    }
+
+    guard let client = try await api.client else {
+      throw MpcError.clientInformationUnavailable
+    }
+
+    var backupSharePairId: String? = nil
+    var walletId: String? = nil
+
+    for wallet in client.wallets {
+      if wallet.curve == .ED25519 {
+        for backupShare in wallet.backupSharePairs {
+          if backupShare.status == .completed, backupShare.backupMethod == method {
+            backupSharePairId = backupShare.id
+            walletId = wallet.id
+            break
+          }
+        }
+      }
+    }
+
+    guard let backupSharePairId = backupSharePairId else {
+      throw MpcError.unableToEjectWallet("No backed up share pair found for curve ED25519.")
+    }
+    guard let walletId = walletId else {
+      throw MpcError.unableToEjectWallet("No backed up wallet found for curve ED25519.")
+    }
+
+    let backupWithPortal = client.environment?.backupWithPortalEnabled ?? false
+    if backupWithPortal {
+      cipherText = try await api.getClientCipherText(backupSharePairId)
+      organizationShare = try await api.prepareEject(walletId, method)
+    }
+
+    guard let cipherText = cipherText else {
+      throw MpcError.noBackupCipherTextFound
+    }
+    guard let organizationShare = organizationShare else {
+      throw MpcError.noOrganizationShareFound
+    }
+
+    let decryptionKey = try await storage.read()
```

Comment:
this line will return the password, and delete it.
The same line is used for ETH eject, and Solana eject
So on the client side in order to eject both, we have to prompt the user to enter the password twice. because once the ETH eject is called the password will become `nil` and when calling the Solana eject it will not work because the password is `nil`